kml.cc

```diff
@@ -541,7 +541,8 @@ kml_wr_deinit()
 
   if (!posnfilenametmp.isEmpty()) {
 #if __WIN32__
-    MoveFileExA(qPrintable(posnfilenametmp), qPrintable(posnfilename),
+    MoveFileExW((const wchar_t*) posnfilenametmp.utf16(),
```

Comment:
Tam, I think this has been through some mechanical changes without testing. This code is to do an atomic rename so that the freshly written KML (that ends in </kml> is always readable.  Is this MoveFileExW() needed at all? That QFile::rename below should do the trick everywhere, right?

This chunk predates having QtCore.

Reply:
Ugg.  I agree with:
> I think this has been through some mechanical changes without testing.

There are bugs here.  
1. I introduced one with 15f8b8b6.  
See https://bugreports.qt.io/browse/QTBUG-2082
 >                   // QFile.rename will always fail if the target
 >                   // file exists. Posix rename wouldn't
2. It seems to me that on windows when we use MoveFileEx* there is no need to do the QFile::rename, and it is guaranteed to fail as posnfilename will exist due to MoveFileEx*.

The bug report makes it sound like QSaveFile does what we want.  The QSaveFile doc says
>QSaveFile creates a temporary file in the same directory as the final file and atomically renames it. However this is not possible if the directory permissions do not allow creating new files. In order to preserve atomicity guarantees, open() fails when it cannot create the temporary file.

Reply:
I think QSaveFile would cause some ripples that we may not want to hold the PR for.  How about
#if __WIN32__
    MoveFileExW((const wchar_t*) posnfilenametmp.utf16(),
                (const wchar_t*) posnfilename.utf16(),
                MOVEFILE_REPLACE_EXISTING);
#else
    rename(qPrintable(posnfilenametmp), qPrintable(posnfilename));
#endif